src/contexts/PredictoorsContext.tsx

```diff
@@ -338,6 +341,15 @@ export const PredictoorsProvider: React.FC<TPredictoorsContextProps> = ({
     predictedEpochs.current[contractAddress].push(item)
   }, [])
 
+  function arraysAreSame(x: any, y: any) {
```

Comment:
The bug in your original code is that you're returning false inside the forEach loop, but it won't exit the function. You should set arraysAreSame to false and continue checking other elements if a mismatch is found.

we have to use a for loop to compare elements instead of forEach for more control.

We should avoid to use the "var" keyword.

```typescript
function arraysAreSame(x: any[], y: any[]): boolean {
  if (x.length !== y.length) return false;

  for (let index = 0; index < x.length; index++) {
    if (x[index].address !== y[index].address) {
      return false;
    }
  }

  return true; // If all elements match, return true
}

```